### ISSUE TITLE
AppSync response: One or more parameter values were invalid: An AttributeValue may not contain an empty string

### ISSUE THREAD

**Issue by atomicalsoftwares** [aws-amplify/amplify-android#352]
I'm saving my object and making sure nothing is empty even the non required attributes yet I keep getting this error.

```
04-05 10:13:43.771 5965-6058/ai.deepseed.sandbox W/amplify:aws-datastore: Error ended observation of mutation outbox: 
    com.amplifyframework.datastore.DataStoreException: Failed to publish an item to the network. AppSync response contained errors: [One or more parameter values were invalid: An AttributeValue may not contain an empty string (Service: DynamoDb, Status Code: 400, Request ID: NLBOSEK0HKDCF2R2SIMES3JBA3VV4KQNSO5AEMVJF66Q9ASUAAJG)]
        at com.amplifyframework.datastore.syncengine.MutationProcessor.lambda$null$7(MutationProcessor.java:9)
        at com.amplifyframework.datastore.syncengine.-$$Lambda$MutationProcessor$wPEmB5CfYLR0HYaLc6wslQChvtI.accept(lambda)
        at com.amplifyframework.datastore.appsync.AppSyncClient.lambda$mutation$4$AppSyncClient(AppSyncClient.java:2)
        at com.amplifyframework.datastore.appsync.-$$Lambda$AppSyncClient$nKAFu77xjCuzuWw9lR7faFIdN9U.accept(lambda)
        at com.amplifyframework.api.aws.SingleItemResultOperation$OkHttpCallback.onResponse(SingleItemResultOperation.java:20)
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:504)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
        at java.lang.Thread.run(Thread.java:818)
```

Any pointers?

**Comment by jamesonwilliams** [comment]
Hello again @atomicalsoftwares!

Hm. I haven't encountered this one, yet. It looks like we (AWS) could send a better error message back from AppSync, indicating _which_ parameter it thinks is empy:
```
Failed to publish an item to the network. AppSync response contained errors: [One or more parameter values were invalid: An AttributeValue may not contain an empty string (Service: DynamoDb, Status Code: 400, Request ID:NLBOSEK0HKDCF2R2SIMES3JBA3VV4KQNSO5AEMVJF66Q9ASUAAJG)]
```

Can you provide me with a couple more details?
1. Your current full GraphQL schema used for data modeling (is it still [this one](https://github.com/aws-amplify/amplify-android/issues/343#issuecomment-608354979))?
2. A code snippet of where you are forming the model instance, and providing it to `Amplify.DataStore.save(...)`?

**Comment by atomicalsoftwares** [comment]
Sorry for the late response, redeploying from Afghanistan is a pain.


>     1. Your current full GraphQL schema used for data modeling (is it still [this one](https://github.com/aws-amplify/amplify-android/issues/343#issuecomment-608354979))?

Yes, the schema is still the same.


>     2. A code snippet of where you are forming the model instance, and providing it to `Amplify.DataStore.save(...)`?

```
MyModel.Builder builder = new MyModel.Builder();
builder.id(UUID.randomUUID().toString());
builder.name("string");
builder.data(list.getAdapter()).toJsonArray().toString());
builder.typeId("my" + "-" + "id");
builder.status(true);
builder.model(new JSONObject().toString());
builder.notes("Alpha");
builder.priceRangeStart(0);
builder.priceRangeEnd(0);
builder.merchantId("a unique id);
builder.price("0.00");


Amplify.DataStore.save(builder.build(),
	saved -> runOnUiThread(() -> {
		Toast.makeText(getApplicationContext(), "Changes have been saved", Toast.LENGTH_SHORT).show();
		hideSoftKeyboard();
		finish();
	}),
	saveFailure -> runOnUiThread(() -> Toast.makeText(getApplicationContext(), "There was an error saving the changes", Toast.LENGTH_SHORT).show())
	);
```

> It'd be one of the nullable fields in your model ... `name`, `price`, `notes`, `data`, `model`, `status`, perhaps?

I would definitely agree, but this only happens once the item is saved.

1. Save the query
2. Close the app
3. Open the app
4. Error comes up on it's own without performing any additional queries (I believe it's syncing in the background?)


> (Note: this issue has been raised internally with the AppSync team. The tracking number for the issue is TT#0310343045.)

**Comment by jamesonwilliams** [comment]
Thanks @atomicalsoftwares for following up with the additional info!

> Error comes up on its own without performing any additional queries (I believe it's syncing in the background?)

^ Yes, that surely sounds right. The exception is occurring in the Sync Engine, while the client is trying to publish your locally-saved data back up to AppSync.

I'm not seeing anything obvious from your code snippet. But, I should have enough data now to try and get a repro, using your schema. Will let you know what I find, when I can give this a try...

**Comment by jpignata** [comment]
We've released a 1.0.0 -- can you let us know if you're still having this issue with our current GA version? Thanks!
